### PR TITLE
[Documentation] Warn that DataCollatorForWholeWordMask is limited to BertTokenizer-like tokenizers

### LINKED ISSUE #11768: DataCollatorForWholeWordMask only works for BERT, and nothing is said in the docstring.
## Environment info

- `transformers` version:
- Platform:
- Python version:
- PyTorch version (GPU?):
- Tensorflow version (GPU?):
- Using GPU in script?:
- Using distributed or parallel set-up in script?:

### Who can help
@patrickvonplaten @LysandreJik @patil-suraj @sgugger

## Information

Model I am using (Bert, XLNet ...): DBERTA (V1) BASE

The problem arises when using:
* [x] the official example scripts: (give details below): DataCollatorForWholeWordMask
* [ ] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [ ] my own task or dataset: (give details below)

## To reproduce

The DataCollatorForWholeWordMask, that should be used for pre-training a Roberta model, or a Deberta model for example (as you don't have a SpanCollator), only works for BERT, and one needs to look the details in the collator code to notice this. I've been training a language model from scratch for weeks now, just to notice yesterday that your collator for WholeWordMask is wrong and only works for BERT. 

1. Try to use the DataCollatorForWholeWordMask with any model that is not BERT.

## Expected behavior

A data collator that is included in your data collators should work generally for any model, not only for BERT. Or at least, in the Docstring it should be clear that one will waste huge amounts of money if using this collator for other models that are not BERT. This being said, I would like to know how could I use the word_ids from the tokenizer to do this, as with the TokenClassification example you provide here: https://colab.research.google.com/github/huggingface/notebooks/blob/master/examples/text_classification.ipynb#scrollTo=MOsHUjgdIrIW 
In this example the extension of the token labels doesn't depend on the continuation token having "##" at the beginning, but uses the word ids from the FastTokenizer. I think the DataCollatorForWholeWordMask should work generally, at least for all fast tokenizers, not only for BERT. For my case, I would like to know what can I do to at least train a little bit more with the correct objective, not with normal MLM but with WWMLM.

### PR DESCRIPTION
**Pull request by ionicsolutions**
# What does this PR do?

Currently, the `DataCollatorForWholeWordMasking` added with #7925 only works for one specific family of tokenizers, but the documentation does not mention this nor is the user warned when using this data collator with an incompatible tokenizer. Since the data collator will run with all tokenizers, just not produce the desired output, this is very misleading for users. 

This PR adds a note to the documentation and a warning that is issued when a user attempts to create the whole word mask with a (presumably) incompatible tokenizer.

Fixes #11768

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@sgugger

### CODE REVIEW

File: src/transformers/data/data_collator.py [hunk 1]
```diff
@@ -435,6 +443,11 @@ def _whole_word_mask(self, input_tokens: List[str], max_predictions=512):
         """
         Get 0/1 labels for masked tokens with whole word mask proxy
         """
+        if not isinstance(self.tokenizer, (BertTokenizer, BertTokenizerFast)):
+            warnings.warn(
+                "DataCollatorForWholeWordMask is only suitable for BertTokenizer-like tokenizers."
+                "Please refer to the documentation for more information."
+            )
```
Comment (sgugger) [inline src/transformers/data/data_collator.py:450]: Maybe just raise an error since it's not supposed to be used in that case?

**Review by sgugger — APPROVED** [review]
Thanks a lot for doing this! I should have time in the coming month to produce a new version of that data collator that works for every tokenizer (at least fast tokenizer) but in the meantime, this is very much welcome!

File: src/transformers/data/data_collator.py (hunk 1 above)
Comment (ionicsolutions) [reply src/transformers/data/data_collator.py:450]: I thought about this, but settled for a warning, because I wasn't sure whether there are other tokenizers that do not inherit from `BertTokenizer`/`BertTokenizerFast` but are nevertheless compatible.
